feat(client): translation status banner

scissorsneedfoodtoo opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ ] My pull request targets the `main` branch of freeCodeCamp.
- [ ] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->

Basic working version of the translation status banner.

scissorsneedfoodtoo committed:
feat: translation status banner

nhcarrigan commented:
Whoops - failing test is my fault. Didn't plan for booleans in the translation objects. Will push a commit to resolve.

nhcarrigan commented:
![test](https://user-images.githubusercontent.com/63889819/107259300-b040be80-69f1-11eb-819e-1d5c72c70db0.gif)

On smaller screens, do we want the banner to shift down on menu toggle? Or allow the search bar to overlap?

Commit:
fix(tools): account for boolean in test

Updates the schema validation step which checks for empty keys to allow
for `false` Boolean values.

scissorsneedfoodtoo commented:
Thanks for catching those things @nhcarrigan, especially that bit with the banner on mobile screens. I can't remember how we used to handle it on News back in the day. Do you remember @ahmadabdolsaheb?

Looks a bit off since it's not quite completely covered. May be better to shift it down with the menu toggle, but that's just me.

Not sure if booleans are the best way to go here, but it was just a fast way to get things working. Feel free to make any necessary changes.

nhcarrigan commented:
Rather than shifting it down (and thus having to shift down the entire page content) what if we shrink it to match the height of the search bar?

![test](https://user-images.githubusercontent.com/63889819/107264118-6eb31200-69f7-11eb-93a6-9d455d47fed3.gif)

Commit:
fix(client): slightly smaller banner

Shrinks the banner a bit so the drop-down search bar overlays it.

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

raisedadead reviewed (commented):
Let's change the banner message to something like:

```
Translations for these parts are in progress, you can help [here].
```

with the link to:

```
https://contribute.freecodecamp.org/#/how-to-translate-files
```

Commit:
feat: modify banner text

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

RandellDawson commented:
When this PR is merged, we need to run the Crowdin UI upload workflow, so @miyaliu666 and @RafaelDavisH can make the translations.

nhcarrigan commented:
@scissorsneedfoodtoo if you want to take another look, I think I'm happy with this.

scissorsneedfoodtoo commented:
@nhcarrigan, that's a great idea. Looks much cleaner that way. I'll make that change now, along with adding a link to the contribute docs.

scissorsneedfoodtoo committed:
fix: update link text and schema, add period

raisedadead commented:
Is this ready for review, we can add the translations for the strings after the fact. If yes, a quick screenshot in here, and we can get this signed-off with Quincy before going live.

nhcarrigan commented:
Sure - give me a moment to spin it up

nhcarrigan commented:
![image](https://user-images.githubusercontent.com/63889819/107322264-0137cd80-6a59-11eb-9b35-dae6d5a04735.png)

@raisedadead here is how it currently looks. I think @scissorsneedfoodtoo is taking one last look at it.

raisedadead commented on client/i18n/locales/english/intro.json:
```diff
@@ -507,6 +518,8 @@
     "expand": "Expand",
-    "collapse": "Collapse"
+    "collapse": "Collapse",
+    "translation-banner": "Translations for this section are in progress.",
```
```suggestion
    "translation-banner": "Help us translate these parts of the curriculum,",
```

raisedadead commented on client/i18n/locales/english/intro.json:
```diff
@@ -507,6 +518,8 @@
-    "collapse": "Collapse"
+    "collapse": "Collapse",
+    "translation-banner": "Translations for this section are in progress.",
+    "translation-link": "You can help"
```
```suggestion
    "translation-link": "here."
```

raisedadead reviewed (changes requested):
Friendlier.

scissorsneedfoodtoo committed:
fix: update banner text based on Quincy's review, change banner to link like on News

raisedadead commented on client/i18n/intro-schema.js:
```diff
@@ -512,7 +523,9 @@ const introSchema = {
     expand: 'Expand',
-    collapse: 'Collapse'
+    collapse: 'Collapse',
+    'translation-banner': 'Translation in progress',
```
```suggestion
    'translation-banner': "Help us translate these parts of the curriculum,",
```

raisedadead commented on client/i18n/intro-schema.js:
```diff
@@ -512,7 +523,9 @@ const introSchema = {
-    collapse: 'Collapse'
+    collapse: 'Collapse',
+    'translation-banner': 'Translation in progress',
+    'translation-link': 'You can help'
```
```suggestion
    'translation-link': "here."
```

scissorsneedfoodtoo committed:
fix: add extra spacing before cert header if not translated

scissorsneedfoodtoo committed:
fix: adjust font-size, padding, color for day/night mode

scissorsneedfoodtoo committed:
fix: adjust font-size, padding, color for day and night mode

scissorsneedfoodtoo commented:
Sorry all, just seeing your comments now. I had a quick call with Quincy a bit earlier to go over the text and updated it based on his input. Other than that made a few stylistic changes and tried to get the functionality closer to the banner over on News.

raisedadead commented:
Sure, no worries, just wanted to make sure we have the message a tad friendly. This looks good. Please mark it ready when done and will merge and deploy.

scissorsneedfoodtoo commented:
Okay, this should be ready for review if anyone wants to take another look.

And here's are some screenshots just in case:

![image](https://user-images.githubusercontent.com/2051070/107327095-dba9d300-6aef-11eb-94e1-b522e977afbc.png)

![image](https://user-images.githubusercontent.com/2051070/107327147-fa0fce80-6aef-11eb-8bae-7f22ddb09776.png)

raisedadead commented on client/i18n/locales/english/intro.json:
```diff
@@ -1,6 +1,7 @@
 {
   "responsive-web-design": {
     "title": "Responsive Web Design",
+    "isTranslated": false,
```
```suggestion
    "isTranslated": true,
```

This should be set to `true`??

scissorsneedfoodtoo replied:
Sorry! Forgot to set that back.

scissorsneedfoodtoo committed:
fix: adjust universal nav breakpoits for translation banner wrapping

scissorsneedfoodtoo committed:
fix: flip english isTranslated flag to true

raisedadead reviewed (approved):
The broken cypress test is not related to this PR, I am disabling it in a separate PR.
